pkg/apis/core/validation/validation.go

```diff
@@ -5330,6 +5330,11 @@ func ValidateResourceRequirements(requirements *core.ResourceRequirements, fldPa
 
 		if helper.IsHugePageResourceName(resourceName) {
 			limContainsHugePages = true
+			if !helper.IsHugePageResourceValueValid(resourceName, quantity) {
+				limContainsHugePages = false
+				allErrs = append(allErrs, field.Invalid(fldPath, quantity.String(),
+					fmt.Sprintf("value must be integer multiple of %s limit", resourceName)))
```

Comment:
if resourceName is "hugepages" the error would be "value must be integer multiple of hugepages limit". This is hard to understand from my point of view. Would it be better to somehow use quantity in the error message/

Reply:
I updated the error msg  below.